packages/strapi-admin/admin/src/containers/Webhooks/EditView/index.js

```diff
@@ -0,0 +1,368 @@
+      disabled: triggerActionIsDisabled,
+      label: formatMessage({
+        id: `Settings.webhooks.trigger`,
+      }),
+      onClick: () => {
+        handleTrigger();
+      },
+      style: {
+        padding: '0 15px',
+      },
+      title: triggerActionIsDisabled
+        ? formatMessage({
+            id: `Settings.webhooks.trigger.save`,
+          })
+        : null,
+      type: 'button',
+      icon: (
+        <Play
+          width="6px"
+          height="7px"
+          fill={triggerActionIsDisabled ? '#b4b6ba' : '#ffffff'}
+        />
+      ),
+    },
+    {
+      color: 'cancel',
+      disabled: actionsAreDisabled,
+      label: formatMessage({
+        id: `app.components.Button.reset`,
+      }),
+      onClick: () => handleReset(),
+      style: {
+        padding: '0 20px',
+      },
+      type: 'button',
+    },
+    {
+      color: 'success',
+      disabled: actionsAreDisabled,
+      label: formatMessage({
+        id: `app.components.Button.save`,
+      }),
+      style: {
+        minWidth: 140,
+      },
+      type: 'submit',
+    },
+  ];
+
+  const headerProps = {
+    title: {
+      label: headerTitle,
+    },
+    actions: actions,
+  };
+
+  const handleBlur = () => {
+    if (submittedOnce) checkFormErrors();
+  };
+
+  const handleChange = ({ target: { name, value } }) => {
+    dispatch({
+      type: 'ON_CHANGE',
+      keys: name.split('.'),
+      value,
+    });
+  };
+
+  const handleClick = () => {
+    dispatch({
+      type: 'ADD_NEW_HEADER',
+      keys: ['headers'],
+    });
+  };
+
+  const handleRemove = ({ event, index }) => {
+    dispatch({
+      type: 'ON_HEADER_REMOVE',
+      index,
+      event,
+    });
+    resetError('headers');
+  };
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    setSubmittedOnce(true);
+    checkFormErrors(true);
+  };
+
+  const submitForm = () => {
+    if (!isCreating) {
+      updateWebhook();
+    } else {
+      createWebhooks();
+    }
+  };
+
+  const checkFormErrors = async (submit = false) => {
+    const webhookToCheck = modifiedWebhook;
+    set(webhookToCheck, 'headers', cleanHeaders());
+
+    try {
+      await createYupSchema(form).validate(webhookToCheck, {
+        abortEarly: false,
+      });
+
+      setErrors({});
+      if (submit) submitForm();
+    } catch (err) {
+      setErrors(getYupInnerErrors(err));
+      if (submit) strapi.notification.error('notification.form.error.fields');
+    }
+  };
+
+  const cleanHeaders = () => {
+    const { headers } = modifiedWebhook;
+
+    if (Object.keys(headers).length === 1) {
+      const { key, value } = headers[0];
+      if (key.length === 0 && value.length === 0) return [];
+    }
+    return headers;
+  };
+
+  const goBack = () => push('/settings/webhooks');
+
+  const resetError = name => {
+    const errors = formErrors;
+
+    if (errors[name]) {
+      delete errors[name];
+      setErrors(errors);
+    }
+  };
+
+  const setErrors = errors => {
```

Comment:
I'm not sure to understand what this function does exactly. From reading the code I have the feeling that it generates and `Object` from the `errors` and it just formats the headers errors but I don't understand why do you need this logic in the code since `yup` will give you and error object with keys like `headers.0.value: { id: 'this value is required }` I think that you can easily pass this error to the input since you have the complete path.

Either way, I think that such functions should be a util so you can easily add tests so we clearly understand what are the expected results.

Reply:
We use the input name to display each error and yup gives me several errors like

```
[
  name: {id: 'This value is required'},
  headers.0.value: { id: 'This value is required'},
  headers.2.value: { id: 'This value is required'}
]
```

so whatever I have to get this array, filter it by keys which contains `headers` and send the array to the input to be split to each subinputs. It's why I made it earlier, so I can have a clean object.